platform/macos/SCsub

```diff
@@ -31,3 +31,6 @@ prog = env.add_program("#bin/godot", files)
 
 if env["debug_symbols"] and env["separate_debug_symbols"]:
     env.AddPostAction(prog, run_in_subprocess(platform_macos_builders.make_debug_macos))
+
+# FIXME: Investigate how to prevent global namespace "pi" from breaking shadowing warnings
+env.Append(CCFLAGS=["-Wno-shadow"])
```

Comment:
CC @bruvzg

Reply:
Specifically:
```
./core/object/method_bind.h:698:17: error: declaration shadows a variable in the global namespace [-Werror,-Wshadow]
                        PropertyInfo pi;
                                     ^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/fp.h:1191:23: note: previous declaration is here
extern const double_t pi                                             __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
```

Reply:
There is `-Wshadow=local` will test if it works on this compiler, it does not support `-Wshadow-local`

Reply:
`-Wshadow=local` is already what we used with GCC prior to this PR, but it only catches local variables shadowing each other. While to match the MSVC check, we also want to catch local variables shadowing member variables.

Reply:
Testing that out locally how to acomplish it in just `platform/macos`

Reply:
Indeed, not sure what accomplishes that on clang++

Reply:
Will test `-Wshadow-uncaptured-local` and some others

Reply:
Testing with some different flags on clang++ and seems to work so far, will see if any more shadowings pop up